os/binfmt/libelf/libelf_addrenv.c

```diff
@@ -77,6 +77,51 @@
 /****************************************************************************
  * Private Functions
  ****************************************************************************/
+#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
+static void allocateregions(FAR struct elf_loadinfo_s *loadinfo)
+{
+	uint32_t totalsize;
+	size_t textsize, datasize, rosize;
+
+	textsize = loadinfo->textsize;
+	rosize = loadinfo->rosize;
+	datasize = loadinfo->binp->ramsize;
+
+	totalsize = textsize + rosize + datasize;
+
+	if (textsize >= rosize && textsize >= datasize) {
```

Comment:
Could be done in short?
something like this: -

	`uint8_t arr[3] = {textsize, rosize, datasize};
	uint8_t *ptr1 = loadinfo->textalloc, *ptr2 = loadinfo->roalloc, *ptr3 = loadinfo->dataalloc;
	
		
	if (arr[0] < arr[1]) {
		<swap arr[0] and arr[1]>
		<swap ptr1 and ptr2>
	}
	
	if (arr[1] < arr[2]) {
		<swap arr[1] and arr[2]>
		<swap ptr2 and ptr3>
		if (arr[0] and arr[1]) {
			<swap arr[0] and arr[1]>
			<swap ptr1 and ptr2>
		}
	}
	
	loadinfo->binp->ramstart = ptr1 = (uintptr_t)kumm_memalign(arr[0], totalsize);

	ptr2 = ptr1 + arr[0];
	ptr3 = ptr2 + arr[1];
	
	<free pointers and array space>`

Reply:
Thanks @amogh-samsung . Done. 
@jeongchanKim I modified as you suggested to use array.